File: sdks/go/pkg/models/operations/putcommentsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PutCommentsIDPathParams struct {
	CommentID string `pathParam:"style=simple,explode=false,name=comment_id"`
}

type PutCommentsIDQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type PutCommentsIDRequestBody struct {
	Message *string `json:"message,omitempty"`
}

type PutCommentsIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  PutCommentsIDPathParams
	QueryParams PutCommentsIDQueryParams
	Request     *PutCommentsIDRequestBody `request:"mediaType=application/json"`
}

type PutCommentsIDResponse struct {
	ClientError *shared.ClientError
	Comment     *shared.Comment
	ContentType string
	StatusCode  int64
}

File: sdks/ts/src/sdk/models/operations/putcommentsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutCommentsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=comment_id" })
  commentId: string;
}


export class PutCommentsIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class PutCommentsIdRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message?: string;
}


export class PutCommentsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutCommentsIdPathParams;

  @SpeakeasyMetadata()
  queryParams: PutCommentsIdQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutCommentsIdRequestBody;
}


export class PutCommentsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  comment?: shared.Comment;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
